components/IscnUploadForm.vue

```diff
@@ -490,57 +490,59 @@ export default class UploadForm extends Vue {
-          type: 'image/jpeg',
-        })
-
-        const fileBytes = (await fileToArrayBuffer(
-          coverFile,
-        )) as unknown as ArrayBuffer
-        if (fileBytes) {
-          const [
-            fileSHA256,
-            imageType,
-            ipfsHash,
-            // eslint-disable-next-line no-await-in-loop
-          ] = await Promise.all([
-            digestFileSHA256(fileBytes),
-            readImageType(fileBytes),
-            Hash.of(Buffer.from(fileBytes)),
-          ])
-
-          const fileRecord: any = {
-            fileName: coverFile.name,
-            fileSize: coverFile.size,
-            fileType: coverFile.type,
-            fileBlob: coverFile,
-            ipfsHash,
-            fileSHA256,
-            isFileImage: !!imageType,
-          }
-
-          epubMetadata.ipfsHash = ipfsHash
+      const coverUrl = await book.coverUrl()
+      if (coverUrl) {
+        const response = await fetch(coverUrl)
+        const blobData = await response.blob()
+        if (blobData) {
+          const coverFile = new File(
+            [blobData],
+            `${metadata.title}_cover.jpeg`,
+            {
+              type: 'image/jpeg',
+            },
+          )
+          const fileBytes = (await fileToArrayBuffer(
+            coverFile,
+          )) as unknown as ArrayBuffer
+          if (fileBytes) {
+            const [
+              fileSHA256,
+              imageType,
+              ipfsHash,
+              // eslint-disable-next-line no-await-in-loop
+            ] = await Promise.all([
+              digestFileSHA256(fileBytes),
+              readImageType(fileBytes),
+              Hash.of(Buffer.from(fileBytes)),
+            ])
 
-          const reader = new FileReader()
-          reader.onload = (e) => {
-            if (!e.target) return
-            fileRecord.fileData = e.target.result as string
+            epubMetadata.ipfsHash = ipfsHash
+            this.epubMetadataList = [
+              ...this.epubMetadataList,
+              epubMetadata,
+            ]
+
+            const fileRecord: any = {
+              fileName: coverFile.name,
+              fileSize: coverFile.size,
+              fileType: coverFile.type,
+              fileBlob: coverFile,
+              ipfsHash,
+              fileSHA256,
+              isFileImage: !!imageType,
+            }
+            const reader = new FileReader()
+            reader.onload = (e) => {
+              if (!e.target) return
+              fileRecord.fileData = e.target.result as string
+              this.fileRecords.push(fileRecord)
+            }
+            reader.readAsDataURL(coverFile)
           }
-          reader.readAsDataURL(coverFile)
-          this.epubMetadataList = [
-            ...this.epubMetadataList,
-            epubMetadata,
-          ]
-          this.fileRecords.push(fileRecord)
         }
+      } else {
+        this.epubMetadataList.push(epubMetadata)
```

Comment:
Do we need `this.epubMetadataList.push(epubMetadata)` also for `if (coverUrl)`? Maybe no need for `else`?

Reply:
The epubMetadata also includes the [title, description, and tags,](https://github.com/likecoin/app-like-co/pull/408/files#diff-b6c5216e116e745da273c1edb6dcccaa190f6d360b269cf1ebe0119d3c6d66d0R466-R473) so it needs to be saved even without a coverUrl.

Reply:
Need to remove `else` then